tests/Datagrid/SimplePagerTest.php

```diff
@@ -34,6 +34,10 @@ public function setUp(): void
             ->getMock();
     }
 
+    /**
+     * @group legacy
+     * @expectedDeprecation The Sonata\AdminBundle\Datagrid\SimplePager::getResults method will not return Doctrine\Common\Collections\ArrayCollection starting from 4.0
```

Comment:
```suggestion
     * @expectedDeprecation Returning a type different than array from Sonata\AdminBundle\Datagrid\SimplePager::getResults is deprecated since sonata-project/admin-bundle 3.x and will return array starting from 4.0.
```

Reply:
👍